api/src/main/java/io/opencensus/stats/MeasureMap.java

```diff
@@ -51,6 +52,17 @@
    */
   public abstract MeasureMap put(MeasureLong measure, long value);
 
+  /**
+   * Associate the contextual information of an {@code Exemplar} to this {@link MeasureMap}.
+   *
+   * <p>If this method is called multiple times, only the last string map will be kept.
+   *
+   * @param attachments contextual information of an {@code Exemplar}.
+   * @return this
+   * @since 0.16
+   */
+  public abstract MeasureMap withAttachments(Map<String, String> attachments);
```

Comment:
This method should be non-abstract (no-op), to avoid breaking any existing subclasses.

Reply:
Done, though I'm wondering should we rather just break other subclasses and force them to implement this method?

Reply:
Do we have any process for making this become abstract at some point?

Reply:
We would need wait at least 1.5 years and increment the major version.

Reply:
Actually, we could make the method abstract sooner if necessary, since we haven't released 1.0 yet, but I would rather keep it non-abstract for longer if we don't have a reason to change it.  We should at least wait one release cycle so that users are able to migrate.

Reply:
Sounds good to me, I'll leave a TODO to make this method abstract in the 0.17 release.

Reply:
sgtm